### PR TITLE
Remove `exclude_paths` support

### PR DESCRIPTION
**Pull request by wfleming**
We no longer provide these in any config visible to engines.

@codeclimate/review

### CODE REVIEW

File: bin/eslint.js [hunk 1]
```diff
@@ -168,12 +145,9 @@ runWithTiming("engineConfig", function () {
...
         engineConfig.include_paths
       );
-    } else if (engineConfig.exclude_paths) {
-      var ignores = engineConfig.exclude_paths;
-      buildFileList = exclusionBasedFileListBuilder(ignores);
     } else {
-      // No includes or excludes, let's try with everything
-      buildFileList = exclusionBasedFileListBuilder([]);
+      // No explicit includes, let's try with everything
+      buildFileList = inclusionBasedFileListBuilder(["./"]);
```
Comment (ABaldwinHunter) [inline bin/eslint.js]: Not sure if analyzing all files in absence of include_paths is the desired behavior? Would there be any cases where no include paths get passed, e.g. calling `analyze` on an excluded directory?

**Comment by ABaldwinHunter** [comment]
One question, otherwise LGTM! Nice to see the excludes go

File: bin/eslint.js [hunk 2]
```diff
@@ -105,31 +105,8 @@ function prunePathsWithinSymlinks(paths) {
...
-        }
-      }
-    });
-
-    return analysisFiles;
-  };
-}
-
 function inclusionBasedFileListBuilder(includePaths) {
+  process.stderr.write("inclusion file builder with include paths: " + includePaths + "\n");
```
Comment (GordonDiggs) [inline bin/eslint.js]: should we wrap this in an `if (debug)` check?

Comment (wfleming) [inline bin/eslint.js]: Yes we should.

File: bin/eslint.js (hunk 1 above)
Comment (wfleming) [inline bin/eslint.js]: This has been the default behavior in many of our engines, including this one, for a while: there are no production instances where it should occur, but it continues to seem like a reasonable default behavior in the absence of anything explicit.

File: bin/eslint.js (hunk 2 above)
Comment (wfleming) [inline bin/eslint.js]: Opted to just remove it, actually. Not sure it's useful longer term.

**Pull request merged**